tracer/src/Datadog.Trace/Util/StringBuilderCache.cs

```diff
@@ -52,5 +52,13 @@ public static string GetStringAndRelease(StringBuilder sb)
 
             return result;
         }
+
+        public static void Release(StringBuilder sb)
+        {
+            if (sb.Capacity <= MaxBuilderSize)
+            {
+                _cachedInstance = sb;
+            }
+        }
```

Comment:
nit: The framework version of this calls `Release()` from `GetStringAndRelease()` just to reduce some duplication I guess? Not bothered either way 🙂 

```csharp
        public static string GetStringAndRelease(StringBuilder sb)
        {
            string result = sb.ToString();
            Release(sb);

            return result;
        }
```

Reply:
I've just done the modification as Zach is off and we'll need to merge this to rebase Lucas' PR on top of it.

Reply:
LGTM